src/windows/index.js

```diff
@@ -21,50 +24,46 @@ export function WindowProvider({ children }) {
         }
       }}
     >
-      {children}
-      {stack.map(
-        (
-          {
-            Component,
-            title,
-            width = 800,
-            props,
-            onClose,
-            closable,
-            expandable,
-            draggable,
-            height,
-            styles,
-            spacing = true
-          },
-          index
-        ) => (
-          <Window
-            key={index}
-            sx={{ display: 'flex !important', flex: '1' }}
-            Title={title}
-            controlled={true}
-            onClose={() => {
-              closeWindow()
-              if (onClose) onClose()
-            }}
-            width={width}
-            height={height}
-            expandable={expandable}
-            draggable={draggable}
-            closable={closable}
-            styles={styles}
-            spacing={spacing}
-          >
-            <Component
-              {...props}
-              window={{
-                close: closeWindow
+      <ClearContext.Provider
+        value={{
+          clear(value) {
+            setClear(value)
```

Comment:
1. save current window data at `stack.length - 1` 
2. closeWindow()
3. Extract stack to function in the component scope
4. modify window data such that recordId inside props should be null
5. remove clear state